File: DBMSWinForms/DBMSWinForms/Column.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DBMSWinForms
{
    public class Column
    {
        public string Name;
        public MyType Type;
        public string typeName;

        public Column(string cname, string ctype)
        {
            Name = cname;
            typeName = ctype;

            switch (ctype)
            {
                case "Integer":
                    Type = new MyTypeInteger();
                    break;
                case "Real":
                    Type = new MyTypesReal();
                    break;
                case "Char":
                    Type = new MyTypeChar();
                    break;
                case "String":
                    Type = new MyTypeString();
                    break;
                case "Time":
                    Type = new MyTypeTime();
                    break;
                case "TimeInvl":
                    Type = new MyTypeTimeInvl();
                    break;
                default:
                    Type = new MyTypeString();
                    break;
            }
        }
    }
}

File: RestAPI/src/models/Column.ts
import CustomInteger from "./CustomInteger";
import CustomString from "./CustomString";
import CustomType from "./CustomType";
import CustomTime from "./CustomTime";
import CustomReal from "./CustomReal";
import CustomTimeInvl from "./CustomTimeInvl";
import CustomChar from "./CustomChar";

export class Column {
  name: string;
  type: CustomType;
  typename: string;

  constructor(name: string, typename: string) {
    this.name = name;
    this.typename = typename;
    switch (typename) {
      case "Integer":
        this.type = new CustomInteger();
        break;
      case "Real":
         this.type = new CustomReal();
         break;
      case "Char":
         this.type = new CustomChar();
         break;
       case "Time":
         this.type = new CustomTime();
         break;
       case "TimeInvl":
         this.type = new CustomTimeInvl();
         break;
      default:
        this.type = new CustomString();
        break;
    }
  }
}
